### ISSUE TITLE
about preprocess

### ISSUE THREAD

**Issue by Wangt-CN** [MysteryVaibhav/m-DAN#1]
Hello, Your code is good, and I want to ask about the pre-process of the flickr30k and the caption?

**Comment by MysteryVaibhav** [comment]
Sure, what is your question ?

**Comment by Wangt-CN** [comment]
Thank you for your email, I haven’t seen the preprocess python file in your repo and there hasn’t been something about this in the readme too, so can you tell me that how you preprocess the flikr30k images and captions?

Thank you very much
Best wishes

Wang tan

________________________________
发件人: Vaibhav <notifications@github.com>
发送时间: Monday, July 16, 2018 9:29:28 AM
收件人: MysteryVaibhav/m-DAN
抄送: Wang Tan; Author
主题: Re: [MysteryVaibhav/m-DAN] about preprocess (#1)


Sure, what is your question ?

―

**Comment by Wangt-CN** [comment]
I have searched another repo [https://github.com/JunweiLiang/DualAttentionNetwork](url)
maybe you used this preprocess?

And can i ask for your training result? i mean the probably epoch and loss, because my loss is still very big (17000) after 10 epoch

**Comment by MysteryVaibhav** [comment]
Replying to the earlier query:
The caption pre-processing code is in _util.py_ itself. The steps for pre-processing captions are just lower casing and marking words as unknown if they occur less than 5 times in all the captions.
The image pre-processing code is not in the repo. The code assumes there's a folder which has a numpy array of shape (regions * feature dimension)  extracted from VGGNet or ResNet (or any other network) for every image id in flickr30K.

**Comment by MysteryVaibhav** [comment]
I don't have the logs now, but I remember distinctly that the training loss started at a very high number of around 25000, with r@1 of around 0.001 after the first epoch. If I used Adadelta optimizer, the loss decreased till 1000 or so after 30 epochs. Your loss value seems reasonable after 10 epochs. Try seeing the recall values after each epoch and see if that improves.

**Comment by Wangt-CN** [comment]
ok, i get it. Thank you very much!

**Comment by MysteryVaibhav** [comment]
Most welcome! Closing this.
